Undeclared identifier in Zig standard library

Zig 0.6.0+1e23a3cd9
Main.zig:
```
const std = @import("std");
pub fn main() anyerror!void {
    try std.fs.deleteFileAbsolute("C:\\Users\\dabbo\\a.txt");
    // Workaround:
    // try std.fs.cwd().deleteFile("C:\\Users\\dabbo\\a.txt");
}
```
Error:
```
C:\Users\dabbo\zig\lib\zig\std\fs.zig:1423:54: error: use of undeclared identifier 'DeleteFileError'
pub fn deleteFileAbsolute(absolute_path: []const u8) DeleteFileError!void {
                                                     ^
.\src\main.zig:3:15: note: referenced here
    try std.fs.deleteFileAbsolute("C:\\Users\\dabbo\\a.txt");
              ^
zig_stdlib_error...The following command exited with error code 1:
C:\Users\dabbo\zig\zig.exe build-exe C:\Users\dabbo\zig_stdlib_error\src\main.zig --cache-dir C:\Users\dabbo\zig_stdlib_error\zig-cache --name zig_stdlib_error --cache on

Build failed. The following command failed:
C:\Users\dabbo\zig_stdlib_error\zig-cache\o\ZVzxj0kCSlYjs4poC6m1NoUhMeHlcOYo1F2sx2VK2wjuK-LYmLWBLKYwHR18JEIO\build.exe C:\Users\dabbo\zig\zig.exe C:\Users\dabbo\zig_stdlib_error C:\Users\dabbo\zig_stdlib_error\zig-cache

```
DeleteFileError is in the Dir struct. I'm guessing either the function should use Dir.DeleteFileError or DeleteFileError should be outside of the struct?

--- a/lib/std/fs.zig
+++ b/lib/std/fs.zig
@@ -1420,7 +1420,7 @@ pub fn createFileAbsoluteW(absolute_path_w: [*:0]const u16, flags: File.CreateFl
 /// Asserts that the path is absolute. See `Dir.deleteFile` for a function that
 /// operates on both absolute and relative paths.
 /// Asserts that the path parameter has no null bytes.
-pub fn deleteFileAbsolute(absolute_path: []const u8) DeleteFileError!void {
+pub fn deleteFileAbsolute(absolute_path: []const u8) Dir.DeleteFileError!void {
     assert(path.isAbsolute(absolute_path));
     return cwd().deleteFile(absolute_path);
 }
@@ -1428,13 +1428,13 @@ pub fn deleteFileAbsolute(absolute_path: []const u8) DeleteFileError!void {
 pub const deleteFileAbsoluteC = @compileError("deprecated: renamed to deleteFileAbsoluteZ");
 
 /// Same as `deleteFileAbsolute` except the parameter is null-terminated.
-pub fn deleteFileAbsoluteZ(absolute_path_c: [*:0]const u8) DeleteFileError!void {
+pub fn deleteFileAbsoluteZ(absolute_path_c: [*:0]const u8) Dir.DeleteFileError!void {
     assert(path.isAbsoluteZ(absolute_path_c));
     return cwd().deleteFileZ(absolute_path_c);
 }
 
 /// Same as `deleteFileAbsolute` except the parameter is WTF-16 encoded.
-pub fn deleteFileAbsoluteW(absolute_path_w: [*:0]const u16) DeleteFileError!void {
+pub fn deleteFileAbsoluteW(absolute_path_w: [*:0]const u16) Dir.DeleteFileError!void {
     assert(path.isAbsoluteWindowsW(absolute_path_w));
     return cwd().deleteFileW(absolute_path_w);
 }

--- a/lib/std/fs/test.zig
+++ b/lib/std/fs/test.zig
@@ -111,7 +111,7 @@ test "open file with exclusive nonblocking lock twice (absolute paths)" {
     defer file1.close();
 
     const file2 = fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
-    std.debug.assert(std.meta.eql(file2, error.WouldBlock));
+    std.testing.expectError(error.WouldBlock, file2);
 
     fs.deleteFileAbsolute(filename) catch |err| switch (err) {
         error.FileNotFound => {},

--- a/lib/std/fs/test.zig
+++ b/lib/std/fs/test.zig
@@ -101,7 +101,11 @@ test "create file, lock and read from multiple process at once" {
 }
 
 test "open file with exclusive nonblocking lock twice (absolute paths)" {
-    const filename = "/tmp/zig-test-absolute-paths.txt";
+    const allocator = std.heap.page_allocator;
+
+    const file_paths: [1][]const u8 = .{"zig-test-absolute-paths.txt"};
+    const filename = try fs.path.resolve(allocator, &file_paths);
+    defer allocator.free(filename);
 
     const file1 = try fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
     defer file1.close();

--- a/lib/std/fs/test.zig
+++ b/lib/std/fs/test.zig
@@ -113,10 +113,7 @@ test "open file with exclusive nonblocking lock twice (absolute paths)" {
     const file2 = fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
     std.testing.expectError(error.WouldBlock, file2);
 
-    fs.deleteFileAbsolute(filename) catch |err| switch (err) {
-        error.FileNotFound => {},
-        else => return err,
-    };
+    try fs.deleteFileAbsolute(filename);
 }
 
 const FileLockTestContext = struct {

--- a/lib/std/fs/test.zig
+++ b/lib/std/fs/test.zig
@@ -101,7 +101,7 @@ test "create file, lock and read from multiple process at once" {
 }
 
 test "open file with exclusive nonblocking lock twice (absolute paths)" {
-    const allocator = std.heap.page_allocator;
+    const allocator = std.testing.allocator;
 
     const file_paths: [1][]const u8 = .{"zig-test-absolute-paths.txt"};
     const filename = try fs.path.resolve(allocator, &file_paths);

--- a/lib/std/fs/test.zig
+++ b/lib/std/fs/test.zig
@@ -108,9 +108,9 @@ test "open file with exclusive nonblocking lock twice (absolute paths)" {
     defer allocator.free(filename);
 
     const file1 = try fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
-    defer file1.close();
 
     const file2 = fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
+    file1.close();
     std.testing.expectError(error.WouldBlock, file2);
 
     try fs.deleteFileAbsolute(filename);

--- a/lib/std/fs/test.zig
+++ b/lib/std/fs/test.zig
@@ -100,6 +100,21 @@ test "create file, lock and read from multiple process at once" {
     };
 }
 
+test "open file with exclusive nonblocking lock twice (absolute paths)" {
+    const filename = "/tmp/zig-test-absolute-paths.txt";
+
+    const file1 = try fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
+    defer file1.close();
+
+    const file2 = fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
+    std.debug.assert(std.meta.eql(file2, error.WouldBlock));
+
+    fs.deleteFileAbsolute(filename) catch |err| switch (err) {
+        error.FileNotFound => {},
+        else => return err,
+    };
+}
+
 const FileLockTestContext = struct {
     filename: []const u8,
     pid: if (builtin.os.tag == .windows) ?void else ?std.os.pid_t = null,

